src/Text/Pandoc/Writers/XWiki.hs

```diff
@@ -0,0 +1,256 @@
+vcat :: [Text] -> Text
+vcat = intercalate "\n"
+
+blockListToXWiki :: PandocMonad m => [Block] -> XWikiReader m Text
+blockListToXWiki blocks =
+  fmap vcat $ mapM blockToXWiki blocks
+
+blockToXWiki :: PandocMonad m => Block -> XWikiReader m Text
+
+blockToXWiki Null = return ""
+
+-- TODO: handle this
+blockToXWiki (Div _ blocks) =
+  blockListToXWiki blocks
+
+blockToXWiki (Plain inlines) =
+  inlineListToXWiki inlines
+
+blockToXWiki (Para inlines) = do
+  contents <- inlineListToXWiki inlines
+  return $ contents <> "\n\n"
+
+blockToXWiki (LineBlock lns) =
+  blockToXWiki $ linesToPara lns
+
+blockToXWiki b@(RawBlock f str)
+  | f == Format "xwiki" = return $ pack str
+  | otherwise           = "" <$ report (BlockNotRendered b)
+
+blockToXWiki HorizontalRule = return "\n----\n"
+
+blockToXWiki (Header level _ inlines) = do
+  contents <- inlineListToXWiki inlines
+  let eqs = Text.replicate level "="
+  return $ eqs <> " " <> contents <> " " <> eqs <> "\n"
+
+-- XWiki doesn't appear to differentiate between inline and block-form code, so we delegate
+-- We do amend the text to ensure that the code markers are on their own lines, since this is a block
+blockToXWiki (CodeBlock attrs str) = do
+  contents <- inlineToXWiki (Code attrs ("\n" <> str <> "\n"))
+  return $ "\n" <> contents <> "\n"
+
+-- TODO: Figure out how to handle this better
+blockToXWiki (BlockQuote blocks) = do
+  blockText <- blockListToXWiki blocks
+  let quoteLines = split (== '\n') blockText
+  let prefixed = map ("> " <>) quoteLines
+  return $ vcat prefixed
+
+blockToXWiki (BulletList contents) = blockToXWikiList "*" $ contents
+
+blockToXWiki (OrderedList _ contents) = blockToXWikiList "1" $ contents
+
+blockToXWiki (DefinitionList items) = do
+  lev <- asks listLevel
+  contents <- local (\s -> s { listLevel = listLevel s <> ";" }) $ mapM definitionListItemToMediaWiki items
+  return $ vcat contents <> if Text.null lev then "\n" else ""
+
+-- TODO: support more features
+blockToXWiki (Table _ _ _ headers rows') = do
+  headers' <- mapM (tableCellXWiki True) headers
+  otherRows <- mapM formRow rows'
+  return $ Data.Text.unlines (Data.Text.unwords headers':otherRows)
+
+formRow :: PandocMonad m => [[Block]] -> XWikiReader m Text
+formRow row = do
+  cellStrings <- mapM (tableCellXWiki False) row
+  return $ Data.Text.unwords cellStrings
+
+
+tableCellXWiki :: PandocMonad m => Bool -> [Block] -> XWikiReader m Text
+tableCellXWiki isHeader cell = do
+  contents <- blockListToXWiki cell
+  let cellBorder = if isHeader then "|=" else "|"
+  return $ cellBorder <> contents
+  
+
+inlineListToXWiki :: PandocMonad m => [Inline] -> XWikiReader m Text
+inlineListToXWiki lst =
+  fmap Data.Text.concat $ mapM inlineToXWiki lst
+
+
+inlineToXWiki :: PandocMonad m => Inline -> XWikiReader m Text
+
+inlineToXWiki (Str str) = return $ escapeXWikiString $ pack str
+
+inlineToXWiki Space = return " "
+
+inlineToXWiki LineBreak = return "\n"
+
+-- FIXME: better understand this
```

Comment:
We can't render a SoftBreak as a newline, because a newline is interpreted as a hard break in xwiki -- or so I gather from your other code.  Hence it becomes a space.